### PR TITLE
Add FXIOS-7097 [v118] Reading list coordinator

### PR DESCRIPTION
**Pull request by FilippoZazzeroni**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7097)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15771)

## :bulb: Description
Add `ReadingListCoordinator` and related tests. The implementation of this coordinator may seems redundant, but it gives space for future navigation in the `ReadingListPanel`, if you think it's not needed let me now, i will close this PR.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by adudenamedruby** [comment]
LGTM and I approve, but @lmarceau, you have the most context for coordinators, so I'll let you give a final approval.

**Review by lmarceau — APPROVED** [review]
🥇

**Comment by lmarceau** [comment]
The build failed due to the following error, can you look into it? 🙏

```
❌  error: Build input file cannot be found: '/Users/vagrant/git/MockLibraryCoordinatorDelegate.swift'. Did you forget to declare this file as an output of a script phase or custom build rule which produces it? (in target 'ClientTests' from project 'Client')
```

**Comment by FilippoZazzeroni** [comment]
@lmarceau i m going to look at it later

**Comment by FilippoZazzeroni** [comment]
Merge on main done

File: Client.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -10442,7 +10457,11 @@
 				28CE83EF1A1D246900576538 /* Third-Party Source */,
 				E15A8B2B286CFD48004BB2B8 /* Tests */,
 				43BE578B278BA4D900491291 /* RustMozillaAppServices-Info.plist */,
+<<<<<<< HEAD
 				E1FC23EF2A86290E0089E14D /* Recovered References */,
+=======
+				C2446B2E2A856BEA000C527D /* Recovered References */,
+>>>>>>> 44f253982 (Add MockLibraryCoordinatorDelegate)
```
Comment (lmarceau) [inline Client.xcodeproj/project.pbxproj:10464]: There's some conflicts unresolved it seems 👀 I haven't checked the other files but please make sure they're all resolved

Comment (FilippoZazzeroni) [reply Client.xcodeproj/project.pbxproj:10464]: Yes I thought there were solved, I'll remove them tomorrow.

Comment (FilippoZazzeroni) [reply Client.xcodeproj/project.pbxproj]: Now the app builds, the conflicts are resolved

**Pull request merged**